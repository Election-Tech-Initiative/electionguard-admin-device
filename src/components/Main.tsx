import styled from 'styled-components'

interface Props {
  padded?: boolean
  noOverflow?: boolean
}

const Main = styled('main')<Props>`
  display: flex;
  flex-direction: column;
  overflow: ${({ noOverflow = false }) => (noOverflow ? undefined : 'auto')};
  padding: ${({ padded = false }) => (padded ? '1rem 0.5rem 2rem' : undefined)};
  @media (min-width: 480px) {
    padding: ${({ padded = false }) => (padded ? '1rem' : undefined)};
  }
`

interface ChildProps {
  center?: boolean
  centerVertical?: boolean
  centerHorizontal?: boolean
  maxWidth?: boolean
}

export const MainChild = styled('div')<ChildProps>`
  margin: ${({
    center = false,
    centerVertical = center,
    centerHorizontal = center,
  }) => `${centerVertical ? 'auto' : '0'} ${centerHorizontal ? 'auto' : '0'}`};
  max-width: ${({ maxWidth = true }) => (maxWidth ? '35rem' : undefined)};
`

export default Main
