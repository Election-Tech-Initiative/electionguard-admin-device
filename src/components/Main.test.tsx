import React from 'react'
import { render } from '@testing-library/react'

import Main, { MainChild } from './Main'

describe('renders Main', () => {
  it('with child', () => {
    const { container } = render(
      <Main>
        <MainChild>foo</MainChild>
      </Main>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('with centered content', () => {
    const { container } = render(
      <Main>
        <MainChild center>foo</MainChild>
      </Main>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('with not centered content', () => {
    const { container } = render(
      <Main noOverflow>
        <MainChild centerHorizontal={false}>foo</MainChild>
      </Main>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('with padding', () => {
    const { container } = render(
      <Main padded>
        <MainChild>padded</MainChild>
      </Main>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
