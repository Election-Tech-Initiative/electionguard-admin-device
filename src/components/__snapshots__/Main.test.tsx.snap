// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Main with centered content 1`] = `
<main
  class="sc-bdVaJa iiFDUH"
>
  <div
    class="sc-bwzfXH jAEsws"
  >
    foo
  </div>
</main>
`;

exports[`renders Main with child 1`] = `
<main
  class="sc-bdVaJa iiFDUH"
>
  <div
    class="sc-bwzfXH cJzayE"
  >
    foo
  </div>
</main>
`;

exports[`renders Main with not centered content 1`] = `
<main
  class="sc-bdVaJa ytQaW"
>
  <div
    class="sc-bwzfXH cJzayE"
  >
    foo
  </div>
</main>
`;

exports[`renders Main with padding 1`] = `
<main
  class="sc-bdVaJa gjMqtI"
>
  <div
    class="sc-bwzfXH cJzayE"
  >
    padded
  </div>
</main>
`;
